### PR TITLE
Fix permissions for /var/lib/rstudio-server directory for sqlite 

### PR DESCRIPTION
**Pull request by jeffvroom**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/9715

### Approach

The sqlite database requires that write perms be available for the directory. Other parts of the code do this for Workbench but OS is missing some code to try and do the chown of the directory so I added it. 

### Automated Tests

Not necessary

### QA Notes

A clean install on a linux machine of the OS version should show /var/lib/rstudio-server owned and writable by rstudio-server

### Checklist

- [x ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ x] This PR passes all local unit tests

### CODE REVIEW

File: src/cpp/server_core/ServerDatabase.cpp [hunk 1]
```diff
@@ -151,7 +151,23 @@ Error readOptions(const std::string& databaseConfigFile,
          // always ensure the database file user is correct, if specified
          error = databaseFile.changeOwnership(databaseFileUser.get());
          if (error)
+         {
+            error.addProperty("description", "Unable to change ownership of database file");
```
Comment (MariaSemple) [inline src/cpp/server_core/ServerDatabase.cpp:155]: Should we include which user we're trying to change the owner to?

File: src/cpp/server_core/ServerDatabase.cpp [hunk 2]
```diff
@@ -151,7 +151,23 @@ Error readOptions(const std::string& databaseConfigFile,
...
             return error;
+         }
+
+#ifndef _WIN32
+         error = databaseDirectory.changeOwnership(databaseFileUser.get());
```
Comment (MariaSemple) [inline src/cpp/server_core/ServerDatabase.cpp:160]: on non windows platforms, this line is duplicated (on successful tests) Should we move the `bool writable = false ...` block inside the first `if (error)` block and avoid calling it twice?

File: src/cpp/server_core/ServerDatabase.cpp [hunk 3]
```diff
@@ -151,7 +151,23 @@ Error readOptions(const std::string& databaseConfigFile,
...
+         if (error)
+         {
+            bool writable = false;
+            Error writableError = databaseDirectory.isWriteable(writable);
+            if (writableError || !writable)
+            {
+                LOG_ERROR_MESSAGE("Sqllite database directory: " + databaseDirectory.getAbsolutePath() + " must be writable by: " + databaseFileUser.get().getUsername());
```
Comment (MariaSemple) [inline src/cpp/server_core/ServerDatabase.cpp:167]: Shouldn't we exit with an error after logging this message?

**Review by MariaSemple — CHANGES_REQUESTED** [review]

File: src/cpp/server_core/ServerDatabase.cpp (hunk 2 above)
Comment (jeffvroom) [reply src/cpp/server_core/ServerDatabase.cpp:160]: The first changeOwnership call is for databaseFile. This one is for databaseDirectory... due to some weird sqlite requirement that the directory itself be writable.

File: src/cpp/server_core/ServerDatabase.cpp (hunk 3 above)
Comment (jeffvroom) [reply src/cpp/server_core/ServerDatabase.cpp:167]: For OS, it seems like things work pretty well even with a read-only database. I was concerned about adding a new "exit error" that might trip up some users that don't care about the DB.

File: src/cpp/server_core/ServerDatabase.cpp (hunk 2 above)
Comment (MariaSemple) [reply src/cpp/server_core/ServerDatabase.cpp:160]: Ah! Such close names! Makes sense to me!

File: src/cpp/server_core/ServerDatabase.cpp (hunk 3 above)
Comment (MariaSemple) [reply src/cpp/server_core/ServerDatabase.cpp:167]: I see, that seems reasonable.

**Review by MariaSemple — APPROVED** [review]
LGTM!

**Pull request merged**